spec/controllers/games_controller_spec.rb

```diff
@@ -136,4 +136,184 @@
       expect(response).to redirect_to(games_url)
     end
   end
+
+  # Adds white pieces to board
+
+  describe 'populate board' do
+    it 'Adds correct number of white Pawns to the board' do
+      game = Game.create!
+      game.populate_board!
+
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 0)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 1)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 2)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 3)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 4)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 5)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 6)
+      expect(game.pieces.where(type: 'Pawn', color: 'White')).to eq game.pieces.where(position_x: 6, position_y: 7)
```

Comment:
`game.pieces.where(type: 'Pawn', color: 'White')` returns all of the White pawns on the board.

`game.pieces.where(position_x: 6, position_y: 7)` returns all of the pieces at position 6,7.

`where` returns a collection. Not a `single` piece.

I don't think these tests are testing what you think they are.

Could do something like this...

```Ruby
expect(game.pieces.find_by(position_x: 6, position_y: 7).type).to eq 'Pawn'
```

Reply:
You're right! I was not aware they were testing like that. Good to know - adjusting now!